src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+  ) {
+    const { x = 0, y = 0 } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class DeliveryDrone {}
+    this.coords = {
+      x,
+      y,
+    };
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion,) {
+    super(name, weight, coords, chipVersion,);
+
+    const { x = 0, y = 0, z = 0 } = coords;
+
+    this.coords = {
+      x,
+      y,
+      z,
+    };
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null
+  ) {
+    super(name, weight, coords, chipVersion,);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (
+      cargo !== null
+      && this.maxLoadWeight >= cargo.weight
+      && this.currentLoad === null
+    ) {
+      this.currentLoad = cargo;
+    }
+  }
+
+  unhookLoad() {
+    if (this.currentLoad !== null) {
+      this.currentLoad = null;
+    }
```

Comment:
"if" is extra here, you just should set null to currentLoad in any cases
```suggestion
  unhookLoad() {
    this.currentLoad = null;
```
![image](https://user-images.githubusercontent.com/87998640/228358123-fd0de79e-e9bb-47a5-9fe9-ff6b8e19fb08.png)